s3-commons/src/main/java/io/aiven/kafka/connect/s3/S3SourceBaseConfig.java

```diff
@@ -0,0 +1,176 @@
+import static io.aiven.kafka.connect.s3.S3CommonConfig.AWS_STS_ROLE_ARN;
+import static io.aiven.kafka.connect.s3.S3CommonConfig.AWS_STS_ROLE_EXTERNAL_ID;
+import static io.aiven.kafka.connect.s3.S3CommonConfig.AWS_STS_ROLE_SESSION_DURATION;
+import static io.aiven.kafka.connect.s3.S3CommonConfig.AWS_STS_ROLE_SESSION_NAME;
+import static io.aiven.kafka.connect.s3.S3CommonConfig.handleDeprecatedYyyyUppercase;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.common.config.ConfigDef;
+
+import io.aiven.kafka.connect.common.config.SourceCommonConfig;
+import io.aiven.kafka.connect.iam.AwsStsEndpointConfig;
+import io.aiven.kafka.connect.iam.AwsStsRole;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.regions.Regions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+@SuppressWarnings({ "PMD.ExcessiveImports", "PMD.TooManyStaticImports" })
+public class S3SourceBaseConfig extends SourceCommonConfig {
+    public static final Logger LOGGER = LoggerFactory.getLogger(S3SourceBaseConfig.class);
+
+    protected S3SourceBaseConfig(ConfigDef definition, Map<String, String> originals) { // NOPMD UnusedAssignment
+        super(definition, handleDeprecatedYyyyUppercase(originals));
+    }
+
+    protected static void addS3RetryPolicies(final ConfigDef configDef) {
+        S3CommonConfig.addS3RetryPolicies(configDef);
+    }
+
+    protected static void addAwsConfigGroup(final ConfigDef configDef) {
+        S3CommonConfig.addAwsConfigGroup(configDef);
+    }
+
+    protected static void addAwsStsConfigGroup(final ConfigDef configDef) {
+        S3CommonConfig.addAwsStsConfigGroup(configDef);
+    }
+
+    protected static void addDeprecatedConfiguration(final ConfigDef configDef) {
+        S3CommonConfig.addDeprecatedConfiguration(configDef);
+    }
+
+    public AwsStsRole getStsRole() {
+        return new AwsStsRole(getString(AWS_STS_ROLE_ARN), getString(AWS_STS_ROLE_EXTERNAL_ID),
+                getString(AWS_STS_ROLE_SESSION_NAME), getInt(AWS_STS_ROLE_SESSION_DURATION));
+    }
+
+    public boolean hasAwsStsRole() {
+        return getStsRole().isValid();
+    }
+
+    public boolean hasStsEndpointConfig() {
+        return getStsEndpointConfig().isValid();
+    }
+
+    public AwsStsEndpointConfig getStsEndpointConfig() {
+        return new AwsStsEndpointConfig(getString(AWS_STS_CONFIG_ENDPOINT), getString(AWS_S3_REGION_CONFIG));
+    }
+
+    public AwsClientBuilder.EndpointConfiguration getAwsEndpointConfiguration() {
+        final AwsStsEndpointConfig config = getStsEndpointConfig();
+        return new AwsClientBuilder.EndpointConfiguration(config.getServiceEndpoint(), config.getSigningRegion());
+    }
+
+    public BasicAWSCredentials getAwsCredentials() {
```

Comment:
How about moving this method to a common class ? Now this is duplicated in both s3 sink and source ?
There are more of these here.

Reply:
Unfortunately to properly use getPassword,  getString etc they are required to be here, but as they are just accessors to pull the data out of the config I am not worries about these right now.

Reply:
@Claudenw lgtm. If you submit a pr, we can review it.